packages/netlify-cms-core/src/formats/helpers.js

```diff
@@ -1,12 +1,7 @@
 export const sortKeys = (sortedKeys = []) => (a, b) => {
   const idxA = sortedKeys.indexOf(a);
   const idxB = sortedKeys.indexOf(b);
-  if (idxA === -1 || idxB === -1) {
```

Comment:
Any reason to remove this?

```js
yaml.toFile({ title: 'title', unknown1: 'unknown1', body: 'body', unknown2: 'unknown2' }, [
          'title',
          'body',
        ])

yaml.toFile({ title: 'title', body: 'body', unknown1: 'unknown1', unknown2: 'unknown2' }, [
          'title',
          'body',
        ])
```
Will generate different results.

Reply:
In fact, I think we should remove the sort and preserve the original field order.  If there is an unknown field, the field order is rearranged. I will be happy to leave it as is if users don't mind.

Reply:
I was actually thinking the same thing about the sorting as I'm not sure why its there in the first place (maybe @erquhart can help), but as long as we have it, it should be consistent.

Reply:
This is what the sort is for, on new entry if an optional field is left untouched, the value is not add on the frontmatter, on subsequent edits when the field value is entered this sort helps to move it to the right position.

Reply:
It would make sense to do the same for the unknown fields.

Reply:
Oh, I get it now.

Reply:
Ok I think this is good to go with https://github.com/netlify/netlify-cms/commit/4e23668412e6d05ccd61954a9abd9ba3d5e395a7

What do you think?

Reply:
Yes, I think so too.